index.js

```diff
@@ -18,145 +20,91 @@ const dataSchemaJSONLatest = require("./includes/data_schema_json_latest");
+        compareChecksums: false, // whether to enable an assertion to compare checksums and row counts in entity_table_check events to checksums and row counts in BigQuery
+        transformEntityEvents: true, // whether to generate tables that transform entity CRUD events into flattened tables
+        urlRegex: null, // re-2 formatted regular expression to use to identify whether a URL is this service's own URL or an external one. If your service only has one domain name set this to 'www.yourdomainname.gov.uk' (without the protocol). If you have more than one use something like '(?i)(www.domain1.gov.uk|www.domain2.gov.uk|www.domain3.gov.uk)'
+        socialRefererDomainRegex: "(?i)(facebook|twitter|^t.co|linkedin|youtube|pinterest|whatsapp|tumblr|reddit)", // re-2 formatted regular expression to use to work out whether an HTTP referer is a social media site
+        searchEngineRefererDomainRegex: "(?i)(google|bing|yahoo|aol|ask.co|baidu|duckduckgo|dogpile|ecosia|exalead|gigablast|hotbot|lycos|metacrawler|mojeek|qwant|searx|swisscows|webcrawler|yandex|yippy)", // re-2 formatted regular expression to use to work out whether an HTTP referer is a search enginer (regardless of whether paid or organic)
+        funnelDepth: 10, // Number of steps forwards/backwards to analyse in funnels - higher allows deeper analysis, lower reduces CPU usage
+        requestPathGroupingRegex: '[0-9a-zA-Z]*[0-9][0-9a-zA-Z]*', // re2-formatted regular expression to replace with the string 'UID' when grouping request paths
+        attributionParameters: ['utm_source', 'utm_campaign', 'utm_medium', 'utm_content', 'gclid', 'gcsrc'], // list of parameters to extract from the request_query array of structs at the beginning of funnels
+        attributionDomainExclusionRegex: "(?i)(signin.education.gov.uk)", //re2-formatted regular expression to use to detect domain names which should be excluded from attribution modelling - for example, the domain name of an authentication service which 
+        dataSchema: [],
+        dependencies: [], // datasets generated outside dfe-analytics-dataform which should be materialised before datasets generated by dfe-analytics-dataform,
+        assertionDisableDuringDateRanges: [{
+            fromMonth: 7,
+            fromDay: 25,
+            toMonth: 9,
+            toDay: 1
+        }, {
+            fromMonth: 3,
+            fromDay: 29,
+            toMonth: 4,
+            toDay: 14
+        }, {
+            fromMonth: 12,
+            fromDay: 22,
+            toMonth: 1,
+            toDay: 7
+        }], // an array of day or date ranges between which some assertions will be disabled if other parameters are set to disable them. Each range is a hash containing either the integer values fromDay, fromMonth, toDay and toMonth *or* the date values fromDate and toDate. Defaults to an approximation to school holidays each year.
+        ...params
```

Comment:
Out of interest will these be updated each year?

Reply:
We could do I guess? Although term dates are pretty loosely defined anyway (different LAs and MATs run to different terms, and bank holidays and inset days fall in random places) so probably not much value in that.

Teams can always override the defaults if they want to.

Also I hope to dispense with data freshness assertions entirely if we manage to get to reliable enough data load such that the checksums can take their place.